src/listenMqtt.js

```diff
@@ -0,0 +1,460 @@
+    ct: "websocket",
+    mqtt_sid: "",
+    //App id from facebook
+    aid: "219994525426954",
+    st: [],
+    pm: [],
+    dc: "",
+    no_auto_fg: true,
+    gas: null
+  };
+  var cookies = ctx.jar.getCookies("https://www.facebook.com").join("; ");
+
+  //Region could be changed for better ping.
+  var host = 'wss://edge-chat.facebook.com/chat?region=atn&sid=' + sessionID;
+
+  var options = {
+    clientId: "mqttwsclient",
+    protocolId: 'MQIsdp',
+    protocolVersion: 3,
+    username: JSON.stringify(username),
+    wsOptions: {
+      'headers': {
+        'Cookie': cookies,
+        'Origin': 'https://www.facebook.com',
+        'User-Agent': ctx.globalOptions.userAgent,
+        'Referer': 'https://www.facebook.com',
+      },
+      origin: 'https://www.facebook.com',
+      protocolVersion: 13
+    }
+  };
+
+  client = fbconnect.connect(host, options);
+
+  client.on('error', function(err) {
+    log.error(err);
+    client.end();
+  });
+
+  client.on('connect', function() {
+    client.subscribe(["/legacy_web", "/webrtc", "/br_sr", "/sr_res", "/t_ms", "/thread_typing", "/orca_typing_notifications", "/notify_disconnect", "/orca_presence"],
+      (err, granted) => {
+        client.unsubscribe('/orca_message_notifications', (err) => {
+          var queue = {
+            sync_api_version: 10,
+            max_deltas_able_to_process: 1000,
+            delta_batch_size: 500,
+            encoding: "JSON",
+            entity_fbid: ctx.userID,
+            initial_titan_sequence_id: lastSeqId,
+            device_params: null
+          };
+
+          client.publish('/messenger_sync_create_queue', JSON.stringify(queue), {qos: 0, retain: false})
+        });
+      });
+  });
+
+  client.on('message', function(topic, message, packet) {
+    var jsonMessage = JSON.parse(message);
+    if(topic === "/t_ms") {
+      if(jsonMessage.lastIssuedSeqId) {
+        lastSeqId = parseInt(jsonMessage.lastIssuedSeqId);
+      } else {
+        if(jsonMessage.deltas) {
+          lastSeqId = parseInt(jsonMessage.deltas[0].irisSeqId);
+        }
+      }
+
+      //If it contains more than 1 delta
+      for(var i in jsonMessage.deltas) {
+        var delta = jsonMessage.deltas[i];
+        parseDelta(ctx, globalCallback, defaultFuncs, {"delta": delta});
+      }
+    } else if(topic === "/thread_typing" || topic === "/orca_typing_notifications") {
+      var typ = {
+        type: "typ",
+        isTyping: !!jsonMessage.state,
+        from: jsonMessage.sender_fbid.toString(),
+        threadID: utils.formatID((jsonMessage.thread || jsonMessage.sender_fbid).toString())
+      }
+      globalCallback(null, typ);
+    } else if(topic === "/orca_presence") {
+      if(!ctx.globalOptions.updatePresence) {
+        for(var i in jsonMessage.list) {
+          var data = jsonMessage.list[i];
+          var userID = data["u"];
+
+          var presence = {
+            type: "presence",
+            userID: userID,
+            timestamp: data["l"],
```

Comment:
I haven't seen this field in any of my orca_presence messages, but there's one called "c" that's still a mistery to me.

Reply:
Yeah I'm trying to figure out what that is for. I'll update if I found something.